### PR TITLE
FSA22V2-244: Related components section

### PR DESCRIPTION
**Pull request by bsmrdel101**
### Description:

Each detail page should have a section that renders data from the “Related Components” field from Airtable. These will be links to other related components.

Screen Shot 2022-10-25 at 1 28 37 PM

### Spec:

Designs: [Figma](https://www.figma.com/file/UjMQEwUvAsnFAMhYNdUiVT/Apprenticeship-Capstone?node-id=84%3A663)

See Story: [FSA22V2-244](https://sparkbox.atlassian.net/jira/software/c/projects/FSA22V2/boards/125?modal=detail&selectedIssue=FSA22V2-244)

### Validation:

- [x] This PR has code changes, and our linters still pass.
- [x] This PR has new code, so new tests were added or updated, and they pass.

#### To Validate:

1. Make sure all PR Checks have passed (GitHub Actions, CircleCI, Code Climate, etc).
2. Pull down all related branches.
3. In terminal: `npm run dev`.
4. Run `npm run lint`.
5. Run axe Devtools on affected pages.
6. Confirm all tests pass.
7. Go to [localhost:3000](http://localhost:3000/accordian) and scroll down to this component. You should be able to click on any of the links there, and be taken to their relative details page.
8. Verify that the component is responsive to any screen size.
9. Open `src/pages/[id].page.jsx` and delete the contents from relatedComponents on line 8, making it an empty array. The component should no longer be rendered on the page if there's no data.

### CODE REVIEW

**Review by tisidro — CHANGES_REQUESTED** [review]
Hi Bennett, when I shrank it down it did this:
image

I think it needs some responsiveness styling.

**Review by tisidro — DISMISSED** [review]

**Comment by tisidro** [comment]
Changes look great Bennet! :)

File: src/styles/components/_related-components.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
+$breakpoint-phone: 900px;
+
+.rel-component {
+  background-color: map.get($colors, "accent-teal");
+  padding: 6rem 8rem;
```
Comment (changangus) [inline src/styles/components/_related-components.scss:5]: Saw a small issue in mobile that seems to be related to this padding here:
Screen Shot 2022-10-26 at 1 44 40 PM

Changed the values in the dev tools to test what would work, `2rem` was kind of just a random choice but it looks like the side padding should be smaller in mobile view:
Screen Shot 2022-10-26 at 1 45 58 PM

File: src/styles/components/_related-components.scss [hunk 2]
```diff
@@ -1,4 +1,4 @@
-$breakpoint-phone: 600px;
+$breakpoint-phone: 900px;
```
Comment (changangus) [inline src/styles/components/_related-components.scss:1]: 900px is a very large screen width to consider a mobile breakpoint, maybe include a tablet breakpoint and then also add a variable for the mobile breakpoint?

File: src/styles/components/_related-components.scss (hunk 1 above)
Comment (changangus) [inline src/styles/components/_related-components.scss:5]: Saw a small issue in mobile that seems to be related to this padding here:
Screen Shot 2022-10-26 at 1 44 40 PM

Changed the values in the dev tools to test what would work, `2rem` was kind of just a random choice but it looks like the side padding should be smaller in mobile view:
Screen Shot 2022-10-26 at 1 45 58 PM

File: src/components/RelatedComponents.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+export default function RelatedComponents({ components }) {
+  if (!components.length) return '';
+
+  return (
+    <div className="rel-component" data-testid="rel-component">
+      <h2 className="rel-component__heading">
+        <img
+          className="rel-component__heading--arrow"
+          src="cmt_inventory_arrow.svg"
+          alt=""
```
Comment (changangus) [inline src/components/RelatedComponents.jsx:10]: empty alt for decorative image, nice!

**Review by changangus — CHANGES_REQUESTED** [review]
Hey Bennett, logic looked great and everything was working really well, nice work! Just left a couple comments, both related to the mobile view styles of the `RelatedComponents` component.

**Review by changangus — APPROVED** [review]
Hey Bennett, it's looking good and working great! Nice job!

**Pull request merged**